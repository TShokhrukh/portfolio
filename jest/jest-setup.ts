import '@testing-library/jest-dom'
import { TextEncoder, TextDecoder } from 'util'
import { IntersectionObserverMock } from './mock/IntersectionObserverMock '

// @ts-ignore
global.TextEncoder = TextEncoder
// @ts-ignore
global.TextDecoder = TextDecoder
// @ts-ignore
global.IntersectionObserver = IntersectionObserverMock
