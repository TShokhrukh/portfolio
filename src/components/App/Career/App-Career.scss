.App-Career {
  margin-bottom: 100px;
}

.App-CareerContainer {
  max-width: 920px;
}
