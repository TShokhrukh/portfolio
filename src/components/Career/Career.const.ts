import { cn } from '@bem-react/classname'

export const cnCareer = cn('Career')
