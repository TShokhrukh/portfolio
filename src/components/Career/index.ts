export * from './build'
export * from './__types__'
export { Career as CareerBase } from './Career'
