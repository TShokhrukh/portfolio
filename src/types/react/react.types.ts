import type { ElementType } from 'react'

export interface IComponentProp<P = any> {
  component?: ElementType<P>,
}
