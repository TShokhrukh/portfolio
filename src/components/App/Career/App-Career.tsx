import React, { FC } from 'react'
import { Container } from '~/components/Container/build'
import { Career } from '~/components/Career/build'
import { cnApp } from '../App.const'
import { useAppContext } from '../libs/context'
import './App-Career.scss'

export const AppCareer: FC = () => {
  const { careers } = useAppContext()
  return (
    <section id='career' className={cnApp('Career', {})}>
      <Container className={cnApp('CareerContainer', {})}>
        <Career careers={careers} />
      </Container>
    </section>
  )
}
