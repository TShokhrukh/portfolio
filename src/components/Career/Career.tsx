import React, { FC } from 'react'
import { Text } from '@eo-locale/react'
import { BadgeSection } from '~/components/BadgeSection/build'
import { CareerList } from '~/components/CareerList/build'
import { cnCareer } from './Career.const'
import type { ICareerProps } from './__types__'
import './Career.scss'

export const Career: FC<ICareerProps> = ({ careers, className, ...props }) => {
  return (
    <BadgeSection {...props} badge={<Text id='career.badge' />} title={<Text id='career.title' />}
      className={cnCareer({}, [className])}>
      <CareerList careers={careers} className={cnCareer('List')} />
    </BadgeSection>
  )
}
