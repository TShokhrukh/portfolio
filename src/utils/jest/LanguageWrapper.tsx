import React, { ReactNode } from 'react'
import { TranslationsProvider } from '@eo-locale/react'

import type { Locale } from '@eo-locale/core'

interface IWrapperProps {
  children: ReactNode,
  language?: string,
  locales?: Locale[]
}

const defLocales: Locale[] = [
  {
    language: 'en',
    messages: {}
  }
]

export const LanguageWrapper: React.FC<IWrapperProps> = ({
  children,
  language = 'en',
  locales = defLocales
}) => {
  return (
    <TranslationsProvider language={language} locales={locales} onError={() => null}>
      {children}
    </TranslationsProvider>
  )
}
