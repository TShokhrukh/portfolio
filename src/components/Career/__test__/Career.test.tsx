/* global test, expect, describe */
import React from 'react'
import { render, screen } from '@testing-library/react'
import { LanguageWrapper } from '~/utils/jest/LanguageWrapper'
import { Career } from '../Career'
import type { TCareerList } from '../__types__'

describe('Career', () => {
  test('must return the component template', () => {
    const careers: TCareerList = [
      {
        id: '1',
        active: true,
        title: 'Title 1',
        desc: {
          en: 'desc 1'
        },
        startDate: '2011-05-09',
        endDate: '2012-06-04'
      },
      {
        id: '2',
        active: false,
        title: 'Title 2',
        desc: {
          en: 'desc 2'
        },
        startDate: '2020-04-09',
        endDate: '2023-06-04'
      }
    ]
    render(
      <LanguageWrapper>
        <Career careers={careers} data-testid='career' />
      </LanguageWrapper>
    )

    expect(screen.getByTestId('career')).toMatchSnapshot()
  })
})
