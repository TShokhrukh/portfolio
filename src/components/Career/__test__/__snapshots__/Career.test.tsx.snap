// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Career must return the component template 1`] = `
<div
  class="BadgeSection Career"
  data-testid="career"
>
  <div
    class="BadgeSection-Head"
  >
    <span
      class="Badge Badge_color_purple BadgeSection-Badge"
    >
      career.badge
    </span>
    <h3
      class="Typography Typography_color_white Typography_variant_h3 BadgeSection-Title"
    >
      career.title
    </h3>
  </div>
  <div
    class="BadgeSection-Body"
  >
    <div
      class="List CareerList Career-List"
    >
      <div
        class="Card CareerCard CareerCard_active CareerList-Item"
      >
        <h5
          class="Typography Typography_color_white Typography_variant_h5 CareerCard-Title"
        >
          Title 1
        </h5>
        <p
          class="Typography CareerCard-Desc"
        >
          desc 1
        </p>
        <div
          class="CareerCard-Working"
        >
          <p
            class="Typography Typography_color_purple CareerCard-Duration"
          >
            careerCard.durationYears
          </p>
          <p
            class="Typography Typography_color_purple CareerCard-Date"
          >
            2011/May
             - 
            2012/June
          </p>
        </div>
      </div>
      <div
        class="Card CareerCard CareerList-Item"
      >
        <h5
          class="Typography Typography_color_white Typography_variant_h5 CareerCard-Title"
        >
          Title 2
        </h5>
        <p
          class="Typography CareerCard-Desc"
        >
          desc 2
        </p>
        <div
          class="CareerCard-Working"
        >
          <p
            class="Typography Typography_color_purple CareerCard-Duration"
          >
            careerCard.durationYears
            careerCard.durationMonths
          </p>
          <p
            class="Typography Typography_color_purple CareerCard-Date"
          >
            2020/April
             - 
            2023/June
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
